app/views/core/SubscribeModal.coffee

```diff
@@ -56,19 +60,18 @@ module.exports = class SubscribeModal extends ModalView
     return if @state is 'purchasing'
     super(arguments...)
     # NOTE: The PayPal button MUST NOT be removed from the page between clicking it and completing the payment, or the payment is cancelled.
-    if @state is 'choosing-payment-method' and @selectedProduct
-      @renderPayPalButton()
+    @renderPayPalButton()
     null
 
   renderPayPalButton: ->
-    if @$('#paypal-button-container').length
+    if @$('#paypal-button-container').length and not @$('#paypal-button-container').children().length
```

Comment:
Those checks are more to make sure the logic below doesn't fail (from not having an element to attach to), and doesn't replace the button (which restarts all of the paypal setup logic, which it doesn't like to be interrupted sometimes).